node/src/block_ranges.rs

```diff
@@ -444,6 +451,153 @@ impl BlockRanges {
 
         Ok(())
     }
+
+    pub(crate) fn edges(&self) -> BlockRanges {
+        let mut edges = BlockRanges::new();
+
+        for range in self.0.iter() {
+            let start = *range.start();
+            let end = *range.end();
+
+            edges
+                .insert_relaxed(start..=start)
+                .expect("BlockRanges always holds valid ranges");
+            edges
+                .insert_relaxed(end..=end)
+                .expect("BlockRanges always holds valid ranges");
+        }
+
+        edges
+    }
+
+    /// Returns `(left, middle, right)` that can be used for binary search.
```

Comment:
```suggestion
    /// Returns `([left], middle, [right])` that can be used for binary search.
```
suggestion to make the returned type more obvious